hawkular-alerts-engine-extensions/hawkular-alerts-aggregated-events-alerter/src/main/java/org/hawkular/alerts/extensions/ExpressionManager.java

```diff
@@ -0,0 +1,181 @@
+            try {
+                // get all of the triggers tagged for this alerter
+                Collection<Trigger> triggers = definitions.getAllTriggersByTag(TAG_NAME, TAG_VALUE);
+                log.infof("Found [%s] Tag [%s, %s] Triggers!", triggers.size(), TAG_NAME, TAG_VALUE);
+
+                Collection<FullTrigger> activeTriggers = new ArrayList<>();
+                for (Trigger trigger : triggers) {
+                    Collection<Condition> conditions = null;
+                    List<Condition> activeConditions = new ArrayList<>();
+                    try {
+                        if (trigger.isEnabled()) {
+                            conditions = definitions.getTriggerConditions(trigger.getTenantId(), trigger.getId(), null);
+                            log.infof("Checking [%s] Conditions for enabled trigger [%s]!", conditions.size(),
+                                    trigger.getName());
+                        }
+                    } catch (Exception e) {
+                        log.error("Failed to fetch Conditions when scheduling metrics conditions for " + trigger, e);
+                        continue;
+                    }
+                    if (null == conditions) {
+                        continue;
+                    }
+                    for (Condition condition : conditions) {
+                        if (condition instanceof ExternalCondition) {
+                            ExternalCondition externalCondition = (ExternalCondition) condition;
+                            if (TAG_VALUE.equals(externalCondition.getAlerterId())) {
+                                activeConditions.add(externalCondition);
+                            }
+                        }
+                    }
+                    if (!activeConditions.isEmpty()) {
+                        FullTrigger activeTrigger = new FullTrigger();
+                        activeTrigger.setTrigger(trigger);
+                        activeTrigger.setConditions(activeConditions);
+                        activeTriggers.add(activeTrigger);
+                    }
+                }
+                if (!activeTriggers.isEmpty()) {
+                    log.infof("ActiveTriggers: %s", activeTriggers);
+                    cep.updateConditions(activeTriggers);
+                }
+            } catch (Exception e) {
+                log.error("Failed to fetch Triggers for external conditions.", e);
+            }
+        });
+    }
+
+    @Lock(LockType.READ)
+    @Override
+    public TreeSet<Event> processEvents(TreeSet<Event> events) {
+        if (isEmpty(events)) {
+            return events;
+        }
+        List<Event> retained = new ArrayList<>();
+        TreeSet<Event> filtered = new TreeSet<>();
+        for (Event event : events) {
+            if (event.getTags().get(TAG_NAME) != null && event.getTags().get(TAG_NAME).equals(TAG_VALUE)
+                    && event.getContext().get(CONTEXT_PROCESSED) == null) {
```

Comment:
Extensions might generate events as a result of the pre-processing inference.
These events are marked in the context to filter out of the Extension for potential loops.